viewing_party/main.py

```diff
@@ -0,0 +1,227 @@
+        avg_rating = sum/count
+
+    except ZeroDivisionError:
+        avg_rating = 0.0
+    
+    return float(avg_rating)
+
+def get_most_watched_genre(user_data):
+#     #grab all the genres and put into watched_genre_list
+    watched_genre_list = []
+    genre_count = {}
+    most_watched_genre = []
+    current_max = 0
+
+    if not user_data["watched"]:
+        return None
+
+    else:
+        for movie in user_data["watched"]:
+            watched_genre_list.append(movie["genre"])
+    
+#   go through list and count occurences of each genre. store this as key:value pairs in genre_count dictionary
+    
+        for genre in watched_genre_list: 
+            if genre in genre_count:
+                genre_count[genre] = genre_count[genre] +1
+            else:
+                genre_count[genre] = 1
+
+# go through genre_count dictionary and add all the genres with the highest counts into the most_watched_genre list. 
+
+        for val in genre_count.values():
+            if val >= current_max:
+                current_max = val
+    
+        for key, values in genre_count.items():
+            if values == current_max:
+                #winner = key
+                #most_watched_genre.append(winner)
+
+        #return winner
+
+                most_watched_genre.append(key)
+
+        return  ", ".join(most_watched_genre)
+        #return str(most_watched_genre).strip("'[]'")
+
+#************** WAVE 3 ****************
+
+def create_friends_titles_list(user_data):
+    friend_list = user_data["friends"]
+
+    friends_titles_list = []
+
+    for watched_dict in friend_list: 
+        dict_in_friend_list = watched_dict
+        #list of two dictionaries with key["watched"], value is list of dictionaries
+        for movie_dict in dict_in_friend_list["watched"]:
+            movie_dict_copy = movie_dict.copy()
+            friends_titles_list.append(movie_dict_copy)
+
+    return friends_titles_list
+
+
+def get_unique_watched(user_data):
+    my_watched_list = []
+
+    for movie in user_data["watched"]:
+        my_watched_list.append(movie)
+    
+    friends_titles_list = create_friends_titles_list(user_data)
+
+    unique_list = []
+
+    #compare and make list of dictionaries of unique values
+
+    for item in my_watched_list:
+        if item not in friends_titles_list:
+            unique_list.append(item)
+
+
+    return unique_list
+
+def get_friends_unique_watched(user_data):
+    
+    my_watched_list = []
+
+    for movie in user_data["watched"]:
+        my_watched_list.append(movie)
+
+    friends_titles_list = create_friends_titles_list(user_data)
+    #friends_titles_list has all the titles that friends have watched
+
+    friends_watched_list = []
+
+    #make list of dictionaries of unique values
+
+    for item in friends_titles_list:
+        if item not in my_watched_list:
+            if item in friends_watched_list:
+                continue 
+            friends_watched_list.append(item)
```

Comment:
It's slightly clearer to use `not in` instead of a `continue` here:

```suggestion
            if item not in friends_watched_list: 
                friends_watched_list.append(item)
```